pkg/performanceprofile/controller/performanceprofile/status/status.go

```diff
@@ -261,3 +266,52 @@ func removeUnMatchedTunedProfiles(nodes []corev1.Node, profiles []tunedv1.Profil
 	}
 	return filteredProfiles
 }
+
+func CalculateUpdatedStatus(prevStatus *performancev2.PerformanceProfileStatus, profileName string, conditions []conditionsv1.Condition) *performancev2.PerformanceProfileStatus {
```

Comment:
nit: the name of the function is a bit clunky: `status.CalculateUpdatedStatus` but I don't have good suggestions, so this is totally not-blocking (and can be addressed last, or even later)

Reply:
minor: returning an object (not a pointer) signals we are making a copy, especially if we deference it anyway in the call sites; alternatively we can make the receiver an object, not a pointer.

Reply:
> minor: returning an object (not a pointer) signals we are making a copy, especially if we deference it anyway in the call sites; alternatively we can make the receiver an object, not a pointer.

We can change it to an actual object. The only reason I see for keeping it a pointer is because it's easier to check whether the status got updated or not, by checking if the function returned a `nil`.
We can add additional argument for signaling that tough